python/ray/data/dataset.py

```diff
@@ -4218,6 +4218,12 @@ def _aggregate_result(self, result: Union[Tuple, Mapping]) -> U:
         else:
             return result
 
+    @PublicAPI(stability="alpha")
+    @property
```

Comment:
Users expect properties to return immediately. So, if `self.schema()` is slow, users might be surprised.

That said, I think it's reasonable to make `columns` a property. No strong opinion either way.

Reply:
updated to a method...we also need to accept `fetch_if_missing`